imp_developer.py

```diff
@@ -272,56 +326,414 @@ def __base64_encode(str):
+            response = None
+        if not response:
+            response = None
+
+        return response
+
+    @staticmethod
+    def create_deployment(token, devicegroup_id, agent_code, device_code):
+        url = PL_IMPCENTRAL_API_URL_V5 + "deployments"
+        data = ('{"data":{"type":"deployment",'
+              ' "attributes": {'
+              '  "description": "Sublime text"'
+              ', "origin": "sublime"'
+              ', "tags": []'
+              ', "agent_code": ' + json.dumps(agent_code) +
+              ', "device_code" : ' + json.dumps(device_code) +
+              '},'
+              '"relationships": {"devicegroup": {'
+              ' "type": "development_devicegroup", "id": "' + devicegroup_id + '"}}'
+              ' }}')
+        ## create a new deployment
+        response, code = HTTP.post(token, url=url, data=data)
+        error = ImpCentral.handle_http_response(response, code)
+        return response.get("data"), error
+
+    @staticmethod
+    def assign_device(token, devicegroup_id, device_id):
+        url = PL_IMPCENTRAL_API_URL_V5 + "devicegroups/" + devicegroup_id + "/relationships/devices"
+        data = json.dumps({
+                "data" : [{
+                    "type": "device",
+                    "id": device_id
+                }]
+            })
+
+        response, code = HTTP.post(token, url, data)
+        error = ImpCentral.handle_http_response(response, code)
+        return response, error
+
+    @staticmethod
+    def unassign_device(token, devicegroup_id, device_id):
+        url = PL_IMPCENTRAL_API_URL_V5 + "devicegroups/" + devicegroup_id + "/relationships/devices"
+        data = json.dumps({
+                "data" : [{
+                    "type": "device",
+                    "id": device_id
+                }]
+            })
+        # Append the selected device to the device group
+        response, code = HTTP.delete(token, url, data)
+
+        error = ImpCentral.handle_http_response(response, code)
+        return response, error
+
+    @staticmethod
+    def conditional_restart(token, devicegroup_id):
+        url = PL_IMPCENTRAL_API_URL_V5 + "devicegroups/" + devicegroup_id + "/conditional_restart"
+        response, code = HTTP.post(token, url)
 
+        error = ImpCentral.handle_http_response(response, code)
+        return response, error
+
+    @staticmethod
+    def handle_http_response(response, code):
+        if HTTP.is_response_code_valid(code):
+            return None
+
+        # Handle invalid credentials use-case
+        if HTTP.is_invalid_credentials(code, response):
+            return {"code": ImpRequest.INVALID_CREDENTIALS, "error": "Invalid Credentials"}
+
+        # handle wrong input use-case
+        error, title = HTTP.is_wrong_input(code, response)
+        if error:
+            # offer for the user to re-try current action
+            return {"code" : ImpRequest.WRONG_INPUT, "message": error}
```

Comment:
extra spaces around `:`

Reply:
removed, fixed for all file in a codding standard style - like {egg: "123"}